include/bout/array.hxx

```diff
@@ -118,7 +118,15 @@ public:
     swap(*this, other);
   }
 
- /*!
+  /*!
+   * Resize the array to \p new_size
+   */
+  void resize(int new_size) {
```

Comment:
Should <= 0 just indicate a "I'm done with the memory but I'm not letting the Array go out of scope"

Reply:
`int` is correct here, both in terms of how we're internally storing the size and consistency with e.g. `Array::size()`.

The check for `>= 0` should probably be in `Array::get()` (which I note we don't currently check!). Note that `new T[0]` is legal, but `new T[-1]` is not. `new` should really be taking `std::size_t` as well.

Should we just switch to using `std::size_t` internally? The following public members would still have to use `int`:

```cpp
Array(int len); // Constructor -- could provide overload
int size(); // Size query
T& operator[](int ind); // Member access -- could provide overload
```

Reply:
I think we probably should use `size_t` for this, since it is definitely the size of something (the array). Setting the size to zero could correspond to releasing memory, and returning a zero size from `Array::size` can indicate unallocated, so negative values aren't really needed.

Reply:
Ok, I'm going to merge this and change `int` -> `size_t` everywhere in `Array`. I think this should actually be ok as this is the "nice" way round.

Reply:
I'd hold off on that if you can as there are significant implications for vectorisation if I remember correctly